src/pkg/zoci/copier.go

```diff
@@ -5,53 +5,79 @@
 package zoci
 
 import (
-	"bytes"
 	"context"
+	"fmt"
 
-	"github.com/defenseunicorns/pkg/oci"
-	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/zarf-dev/zarf/src/pkg/logger"
-	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2"
+
+	retry "github.com/avast/retry-go/v4"
 )
 
-// CopyPackage copies a zarf package from one OCI registry to another
-func CopyPackage(ctx context.Context, src *Remote, dst *Remote, concurrency int) (err error) {
+// CopyPackage copies a zarf package from one OCI registry to another using ORAS with retry.
+func CopyPackage(ctx context.Context, src *Remote, dst *Remote, opts PublishOptions) (err error) {
 	l := logger.From(ctx)
-	if concurrency <= 0 {
-		concurrency = DefaultConcurrency
-	}
-
-	srcManifest, err := src.FetchRoot(ctx)
-	if err != nil {
-		return err
-	}
-	l.Info("copying package",
-		"src", src.Repo().Reference.String(),
-		"dst", dst.Repo().Reference.String())
-	if err := oci.Copy(ctx, src.OrasRemote, dst.OrasRemote, nil, concurrency, nil); err != nil {
-		return err
+	if opts.OCIConcurrency <= 0 {
+		opts.OCIConcurrency = DefaultConcurrency
```

Comment:
Here, would be the place where I'd default the retries, if they are less than 1.

Reply:
Updated